src/Core/src/Platform/Android/ActivityLifecycleCallbacks.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Threading;
+using Android.App;
+using Android.OS;
+using Microsoft.Maui.LifecycleEvents;
+
+namespace Microsoft.Maui
+{
+	class ActivityLifecycleCallbacks : Java.Lang.Object, Application.IActivityLifecycleCallbacks
+	{
+		int startedCounter = 0;
+		int resumedCounter = 0;
+		TimeSpan timeout = TimeSpan.FromMilliseconds(700);
+		Timer? timer;
+		bool pauseSent = true;
+		bool stopSent = true;
+		IServiceProvider services;
+		Application application;
+
+		public ActivityLifecycleCallbacks(Application application, IServiceProvider services)
+		{
+			this.application = application;
+			this.services = services;
+		}
+
+		public void OnActivityCreated(Activity activity, Bundle? savedInstanceState)
+		{
+			activity.RegisterComponentCallbacks(new ActivityComponentCallbacks(activity, services));
+			services?.InvokeLifecycleEvents<AndroidLifecycle.OnCreate>(del => del(activity, savedInstanceState));
+		}
+
+		public void OnActivityStarted(Activity activity)
+		{
+			OnActivityStarted();
+			services?.InvokeLifecycleEvents<AndroidLifecycle.OnStart>(del => del(activity));
+		}
+
+		public void OnActivityResumed(Activity activity)
+		{
+			OnActivityResumed();
+			services?.InvokeLifecycleEvents<AndroidLifecycle.OnResume>(del => del(activity));
+		}
+
+		public void OnActivityPaused(Activity activity)
+		{
+			OnActivityPaused();
+			services?.InvokeLifecycleEvents<AndroidLifecycle.OnPause>(del => del(activity));
+		}
+
+		public void OnActivityStopped(Activity activity)
+		{
+			OnActivityStopped();
+			services?.InvokeLifecycleEvents<AndroidLifecycle.OnStop>(del => del(activity));
+		}
+
+		public void OnActivitySaveInstanceState(Activity activity, Bundle outState) =>
+			services?.InvokeLifecycleEvents<AndroidLifecycle.OnSaveInstanceState>(del => del(activity, outState));
+
+		public void OnActivityDestroyed(Activity activity) =>
+			services?.InvokeLifecycleEvents<AndroidLifecycle.OnDestroy>(del => del(activity));
+
+		public void OnActivityStarted()
+		{
+			startedCounter++;
+			if (startedCounter != 1 || !stopSent)
+				return;
+			services?.InvokeLifecycleEvents<AndroidLifecycle.OnApplicationStart>(del => del(application));
```

Comment:
This is semi-interesting as this will fire once, and when the app is starting. So is this the same as the CreateMauiApp? This event seems to be tied to "application starts when the first activity is starting" and I am not sure that is correct as you could have an app that starts and does not show UI but still needs to run because reasons.

Maybe I am manufacturing cases. But for maui CORE, I would rather have less that is more accurate 100% than "mostly accurate".